network/peer/peer.go

```diff
@@ -870,35 +881,46 @@ func (p *peer) handleVersion(msg *p2p.Version) {
 		return
 	}
 
-	peerVersion, err := version.ParseApplication(msg.MyVersion)
-	if err != nil {
-		p.Log.Debug("failed to parse peer version",
-			zap.Stringer("nodeID", p.id),
-			zap.Error(err),
-		)
-		p.StartClose()
-		return
+	if msg.Client != nil {
+		p.version = &version.Application{
+			Name:  msg.Client.Name,
+			Major: int(msg.Client.Major),
+			Minor: int(msg.Client.Minor),
+			Patch: int(msg.Client.Patch),
```

Comment:
It feels weird that the p2p type uses uints but we internally use signed integers with the existing type. I think we can change this in a future PR.